tests/test_comps.py

```diff
@@ -166,98 +166,170 @@ def setUp(self):
         somerset = self.persistor.group('somerset')
-        self.assertTrue(somerset.installed)
+        self.assertTrue(somerset.is_installed)
         self.assertEqual(somerset.pkg_types, dnf.comps.MANDATORY)
-        self.assertCountEqual(somerset.pkg_exclude, ('lotus',))
+        self.assertCountEqual(somerset.get_exclude(), ('lotus',))
         base = self.persistor.group('somerset')
-        self.assertTrue(base.installed)
+        self.assertTrue(base.is_installed)
 
     def test_remove(self):
+        # env is installed from previous transaction
         env = self.comps.environment_by_pattern('sugar-desktop-environment')
-        self._install(env)
         trans = self.solver._environment_remove(env.id)
+        self.persistor.commit()
 
         p_env = self.persistor.environment('sugar-desktop-environment')
-        self.assertTransEqual(trans.remove, ('pepper', 'trampoline', 'hole'))
-        self.assertFalse(p_env.grp_types)
-        self.assertFalse(p_env.pkg_types)
+        self.assertFalse(p_env.is_installed())
+        self.assertEqual(p_env.pkg_types, dnf.comps.MANDATORY)
+        self.assertEqual(p_env.grp_types, dnf.comps.ALL_TYPES)
+
+        grp_list = p_env.get_group_list()
+        self.assertTrue(len(grp_list))
+        for grp in grp_list:
+            _grp = self.persistor.group(grp)
+            self.assertEqual(_grp.pkg_types, dnf.comps.MANDATORY)
+            self.assertFalse(_grp.is_installed)
+
+        # install it again with different pkg_types
+        self.solver._environment_install(env.id, dnf.comps.OPTIONAL, [])
+        self.persistor.commit()
+        p_env = self.persistor.environment('sugar-desktop-environment')
+        self.assertTrue(p_env.is_installed())
+        self.assertEqual(p_env.pkg_types, dnf.comps.OPTIONAL)
+        self.assertEqual(p_env.grp_types, dnf.comps.ALL_TYPES)
+        grp_list = p_env.get_group_list()
+        self.assertTrue(len(grp_list))
+        for grp in grp_list:
+            _grp = self.persistor.group(grp)
+            self.assertEqual(_grp.pkg_types, dnf.comps.OPTIONAL)
 
     def test_upgrade(self):
         """Upgrade environment, the one group it knows is no longer installed."""
+        self.history.reset_db()
+        env = self.comps.environment_by_pattern('sugar-desktop-environment')
+        self.solver._environment_install(env.id, dnf.comps.MANDATORY, [])
+        self.persistor.commit()
+
         p_env = self.persistor.environment('sugar-desktop-environment')
-        p_env.full_list.extend(['somerset'])
-        p_env.grp_types = dnf.comps.ALL_TYPES
-        p_env.pkg_types = dnf.comps.ALL_TYPES
+        self.assertTrue(p_env.is_installed())
 
-        env = self.comps.environment_by_pattern('sugar-desktop-environment')
-        trans = self.solver._environment_upgrade(env.id)
+        grp = self.comps.group_by_pattern('Peppers')
+
+        self.solver._group_remove(grp.id)
+        self.persistor.commit()
+
+        trans = self._install(env, False)
```

Comment:
@jsilhan @MichaelMraka 
This should be test for env update but desc. says: `one group it knows is no longer installed`.
Env in swdb is currently considered installed only with all of its groups installed - so update command does fail on "env is no longer installed"...

Is this approach correct?